### ISSUE TITLE
Suggest Fix & Autofix

### ISSUE THREAD

**Issue by bluesign** [onflow/cadence-tools#12]
Analyzers are amazing, it would be great if they also offer the fix. 
For example: with LS and codelens, vs code can guide users to the fix within a click.

Something like: replace `this document range` with `fix` would be very valuable

**Comment by turbolent** [comment]
Good idea. For the linter / the analyzers to be able to provide suggested fixes, we'll need to add support in Cadence to [`analysis.Diagnostic`](https://github.com/onflow/cadence/blob/98e4457cb8889394a82c80d5120effae8ea1b9c1/tools/analysis/diagnostic.go#L26-L32), similar to how diagnostics are defined in Go: https://pkg.go.dev/golang.org/x/tools/go/analysis#SuggestedFix
